File: leetcode_ly/java/Array/SummaryRanges.java
public class Solution {
    public List<String> summaryRanges(int[] nums) {
        ArrayList<String> list = new ArrayList<String>();
        if(nums.length == 0) return list;
        int i = 0;
        while(i < nums.length){
            String s = new String();
            s += nums[i];
            //用来判断首尾一样的情况
            int temp = nums[i];
            i++;
            while(i < nums.length && nums[i] - nums[i - 1] == 1){
                i++;
            }
            s = s + "->" + nums[i - 1];
            if(temp == nums[i - 1]){
                s = new String();
                s += temp;
            }
            list.add(s);
        }
        return list;
    }
}

File: leetcode_ly/c++/Array/SummaryRanges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> res;
        if(nums.size() == 0) return res;
        if(nums.size() == 1) {
            res.push_back(to_string(nums[0]));
            return res;
        }
        int begin = 0;
        int end = 1;
        int index = 0;
        while(index < nums.size()){
            while(nums[end] - nums[begin] == 1 && end < nums.size()){
                begin++;
                end++;
            }
            if(end - index == 1) res.push_back(to_string(nums[index]));
            else{
                string result = to_string(nums[index]) + "->" + to_string(nums[end - 1]);
                res.push_back(result);
            }
            index = end;
            begin = index;
            end = begin + 1;
        }
        return res;
    }
};